tensorboard/webapp/metrics/effects/index.ts

```diff
@@ -267,29 +267,52 @@ export class MetricsEffects implements OnInitEffects {
     ofType(actions.cardPinStateToggled),
     withLatestFrom(
       this.getVisibleCardFetchInfos(),
-      this.store.select(selectors.getEnableGlobalPins)
+      this.store.select(selectors.getEnableGlobalPins),
+      this.store.select(selectors.getShouldPersistSettings)
     ),
-    map(
-      ([
-        {cardId, canCreateNewPins, wasPinned},
-        fetchInfos,
-        enableGlobalPins,
-      ]) => {
-        if (!enableGlobalPins) {
-          return;
-        }
-        const card = fetchInfos.find((value) => value.id === cardId);
-        // Saving only scalar pinned cards.
-        if (!card || card.plugin !== PluginType.SCALARS) {
-          return;
-        }
-        if (wasPinned) {
-          this.savedPinsDataSource.removeScalarPin(card.tag);
-        } else if (canCreateNewPins) {
-          this.savedPinsDataSource.saveScalarPin(card.tag);
-        }
+    filter(
+      ([, , enableGlobalPins, shouldPersistSettings]) =>
+        enableGlobalPins && shouldPersistSettings
+    ),
+    map(([{cardId, canCreateNewPins, wasPinned}, fetchInfos]) => {
+      const card = fetchInfos.find((value) => value.id === cardId);
+      // Saving only scalar pinned cards.
+      if (!card || card.plugin !== PluginType.SCALARS) {
+        return;
+      }
+      if (wasPinned) {
+        this.savedPinsDataSource.removeScalarPin(card.tag);
+      } else if (canCreateNewPins) {
+        this.savedPinsDataSource.saveScalarPin(card.tag);
       }
-    )
+    })
+  );
+
+  private readonly loadSavedPins$ = this.actions$.pipe(
+    ofType(initAction, coreActions.pluginsListingLoaded),
```

Comment:
The requirement is to dispatch `metricsUnresolvedPinnedCardsAdded` before `stateRehydratedFromUrl`, right? `pluginsListingLoaded` doesn't actually guarantee this and is unsafe to use here (we can also verify this with dev console logs).

The metrics effect initAction will be safe to use as all effect initialization will happen in an earlier event loop iteration than `stateRehydratedFromUrl` because of the [delay](https://github.com/tensorflow/tensorboard/blob/c6c3545970457f8532f794848fc2a67cb9c6e375/tensorboard/webapp/app_routing/effects/app_routing_effects.ts#L186) placed before it (btw this file also shows examples of initAction usage). For the current purposes, it might be most appropriate to listen to the App Routing initAction.

Reply:
Thanks for comments! I thought of listening to `App Routing initAction`, but I thought that considering other initActions might be more confusing for maintenance. I think the init action of the metrics effect and leaving a comment that it should be dispatched before `stateRehydratedFromUrl ` will be sufficient for now.